### ISSUE TITLE
Console Permissions - Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by jcmoore0** [hasura/graphql-engine#3789]
**Problem**:  **Error:  Cannot read property 'length' of undefined**
Additional Info:
  1.  Occurs when editing 'SELECT' permission.
  2.  Is intermittent and usually occurs on 1 of my 2 roles for a table.
  3.  Bulk sql/ loadSchema data from server is missing 'permissions.select.computed_fields' empty array
  4.  I don't have any computed_columns

**v1.1.0-beta.2**

**line 89 -  console/src/components/Services/Data/PermissionsSummary/utils.js**
permission['computed_fields'] is undefined
```
 Object.keys(tableFields).forEach(fieldType => {
      noFields = noFields && !permission[fieldType].length;

      allFields =
        allFields &&
        (permission[fieldType] === '*' || 
          permission[fieldType].length === tableFields[fieldType].length);   <-- there is not a computed_fields property on permission
    });
```

Thanks,
John

**Comment by jcmoore0** [comment]
Thanks @rikinsk for the quick response.
As a short term work around,  I can use the previous console version to edit permissions.

**Comment by rikinsk** [comment]
@jcmoore0 I was unfortunately not able to reproduce this. I assume you must be seeing a 500 page shown when the error occurs. Could you share the error stack trace displayed.

**Comment by jcmoore0** [comment]
Hi @rikinsk  - I am getting all 200 responses.   It happens on several tables on one of my 2 roles.

data returns with 200:
```
  return dispatch(requestAction(url, options)).then(
      data => {
        const mergedData = mergeLoadSchemaData(
          JSON.parse(data[0].result[1]),
          data[1],
          JSON.parse(data[2].result[1]),
          JSON.parse(data[3].result[1])
        );

**Missing property computed_fields:[]**
data[1][5].permissions[0].permissions.select {
  filter: {attachment: {…}}
  columns: (5) 
  allow_aggregations: false
}
```

```
Client error:
TypeError: Cannot read property 'length' of undefined
    at http://127.0.0.1:3001/rstatic/dist/main-885638bc67bbc108f8ba.js:408806:171
    at Array.forEach (<anonymous>)
    at getPermissionColumnAccessSummary (http://127.0.0.1:3001/rstatic/dist/main-885638bc67bbc108f8ba.js:408801:30)
    at getColumnSection (http://127.0.0.1:3001/rstatic/dist/main-885638bc67bbc108f8ba.js:524287:160)
    at getEditSection (http://127.0.0.1:3001/rstatic/dist/main-885638bc67bbc108f8ba.js:525120:246)
    at Permissions.render (http://127.0.0.1:3001/rstatic/dist/main-885638bc67bbc108f8ba.js:525223:103)
    at Permissions.hotComponentRender (http://127.0.0.1:3001/rstatic/dist/main-885638bc67bbc108f8ba.js:233559:67)
    at Permissions.proxiedRender (http://127.0.0.1:3001/rstatic/dist/main-885638bc67bbc108f8ba.js:233579:36)
    at finishClassComponent (http://127.0.0.1:3001/rstatic/dist/main-885638bc67bbc108f8ba.js:225044:31)
    at updateClassComponent (http://127.0.0.1:3001/rstatic/dist/main-885638bc67bbc108f8ba.js:224999:24)
```

It is ver

**Comment by rikinsk** [comment]
Thanks for the extra info. I'll take a look at it again.

**Comment by rikinsk** [comment]
closed in fca16b8
